Toggl.Core/Interactors/Sync/CreateUnsyncedDataDumpInteractor.cs

```diff
@@ -0,0 +1,42 @@
+using System;
+using System.Collections.Generic;
+using System.Reactive.Linq;
+using System.Threading.Tasks;
+using Toggl.Core.DataSources;
+using Toggl.Core.DataSources.Interfaces;
+using Toggl.Core.Models;
+using Toggl.Core.Models.Interfaces;
+using Toggl.Shared;
+using Toggl.Storage;
+
+namespace Toggl.Core.Interactors
+{
+    public class CreateUnsyncedDataDumpInteractor : IInteractor<Task<SyncableDataDump>>
+    {
+        private readonly ITogglDataSource dataSource;
+
+        public CreateUnsyncedDataDumpInteractor(ITogglDataSource dataSource)
+        {
+            Ensure.Argument.IsNotNull(dataSource, nameof(dataSource));
+            this.dataSource = dataSource;
+        }
+
+        public async Task<SyncableDataDump> Execute()
+        {
+            var clients = await getUnsyncedItems(dataSource.Clients);
+            var projects = await getUnsyncedItems(dataSource.Projects);
+            var tags = await getUnsyncedItems(dataSource.Tags);
+            var timeEntries = await getUnsyncedItems(dataSource.TimeEntries);
+
+            return new SyncableDataDump(timeEntries, tags, projects, clients);
+        }
+
+        private IObservable<IEnumerable<TThreadsafe>> getUnsyncedItems<TThreadsafe, TDatabase>(
+            IDataSource<TThreadsafe, TDatabase> source)
+            where TDatabase : IDatabaseSyncable
+            where TThreadsafe : TDatabase, IThreadSafeModel
+        {
+            return source.GetAll(p => p.SyncStatus != SyncStatus.InSync);
+        }
```

Comment:
EBM wouldn't hurt here:
```suggestion
        => source.GetAll(p => p.SyncStatus != SyncStatus.InSync);
```